R/install.R

```diff
@@ -396,3 +398,172 @@ build_status_ok <- function(process_log, quiet = FALSE) {
 
   TRUE
 }
+
+install_mingw32_make <- function() {
+  processx::run(
+    "pacman",
+    args = c("-Syu", "mingw-w64-x86_64-make","--noconfirm"),
+    wd = file.path(Sys.getenv("RTOOLS40_HOME"), "usr", "bin"),
+    error_on_status = TRUE
+  )
+  invisible(NULL)
+}
+
+check_compiler_version <- function(compiler = c("g++", "clang++")) {
+
+}
```

Comment:
At first I meant to define it but I later figured it will be challenging, because we cant really know whether the user uses g++ or clang before building cmdstan. We should add a compiler check command to cmdstan makefiles. 
We can use that then. 

Too old compiler is generally not a huge support problem. Nowadays its mostly seen when users move to a cluster, where some popular distributions still come with g++ 4.8.x. But users mostly run cmdstan directly on a cluster.